futurex_openedx_extensions/dashboard/serializers.py

```diff
@@ -266,27 +266,37 @@ def __init__(self, *args: Any, **kwargs: Any):
         self._grading_info: Dict[str, Any] = {}
         self._subsection_locations: Dict[str, Any] = {}
 
-    def collect_grading_info(self, course_ids: list) -> None:
-        """Collect the grading info."""
+        # exam_scores will only be included if there is only one course
+        if self.context.get('course_id'):
+            self.collect_grading_info()
+        else:
+            self.fields.pop('exam_scores', None)
```

Comment:
`fields` is a cached property in DRF `cached_property`, which is not predictable at all. Please pop out `exam_scores` from `requested_optional_field_tags`. That will be the safest way to deal with it
```suggestion
            tags = self.context.get('requested_optional_field_tags')
            if tags:
                tags.pop('exam_scores', None)
```